qiskit/tools/visualization/_matplotlib.py

```diff
@@ -363,9 +363,9 @@ def _draw_regs(self):
         # quantum register
         for ii, reg in enumerate(self._qreg):
             if len(self._qreg) > 1:
-                label = '${}_{{{}}}$'.format(reg.name, reg.index)
+                label = '${}_{{{}}}$'.format(reg.name.name, reg.index)
```

Comment:
nonblocking, as this is a quite non-invasive fix, but maybe this code should be updated so it doesn't call a "register" as "name" anymore. 
The original definition is in the `_registers()` method. Instead of:
```
    def _registers(self, creg, qreg):
        # NOTE: formats of clbit and qubit are different!
        self._creg = []
        for e in creg:
            self._creg.append(Register(name=e[0], index=e[1]))
        self._qreg = []
        for e in qreg:
self._qreg.append(Register(name=e[0], index=e[1]))
```
it could just call it `Register(reg=.., index=..)`.

Reply:
the NOTE there is also obsolete because qreg and creg now have the same format.

Reply:
Yeah I thought about doing the bigger fix and updating that function to fix the name, but with the proximity to the release I figured it was easier to just do the quick thing which was very low risk and come back post release and change things up. I was worried changing name to register there would have unintended side effects elsewhere in the module.

Reply:
sounds good.